src/CFamily/CMake/MacroEvaluationService.cs

```diff
@@ -50,7 +50,9 @@ internal class MacroEvaluationService : IMacroEvaluationService
                                      // The word chars are captured in a group called "prefix".
             "?" +                    // Match the previous group 0 or 1 times i.e. the prefix is optional.
             "(?<name>\\w+)" +        // Capture one or more word consecutive word chars in a group called "name".
-            "}");                    // Match the closing literal.
+            "}",                     // Match the closing literal.
+            RegexOptions.Compiled,
+            Core.RegexConstants.DefaultTimeout);
```

Comment:
Macros are evaluated every time we analyze a C++ file or pre-process a C++ header file, which happens every time the user switches between document windows.

i.e. it can happen a lot, and the regular expression pattern is fixed, so I decided to make the expression `Compiled`.